code/controllers/subsystem/processing/station.dm

```diff
@@ -64,6 +65,19 @@ PROCESSING_SUBSYSTEM_DEF(station)
 			var/datum/station_trait/trait_to_remove = i
 			selectable_traits_by_types[initial(trait_to_remove.trait_type)] -= trait_to_remove
 
+///Adds exclusive station trait based on each weight regardless of count
+/datum/controller/subsystem/processing/station/proc/adds_exclusive_traits()
```

Comment:
why not reuse proc above?

Reply:
because the behaviour to pick a trait between the current system and the new system is different. The new way shouldn't have the max trait count.